DavinciSalute/formbricks
# Feature sc 60681 formbricks implementation requirements logic

## What does this PR do?


Sometimes, the `jumpTo` conditional logic is not sufficient to achieve our goals without forcing us to branch the survey and/or duplicate questions.
Our goal is to control the display of questions based on previous answers, while keeping the question order simple.

This PR introduces a new `requirementsLogic` feature and refactors the `getNextQuestionId` function for improved readability and maintainability. The new requirementsLogic functionality allows for dynamic question visibility based on specified conditions, enhancing the survey's interactivity. Additionally, the `getNextQuestionIdByLogicJump` function has been moved to `logicEvaluator` for better organization.

feat: requirementsLogic option for question



## How should this be tested?


To verify the changes, please perform the following tests:

- Test A: Verify that questions with requirementsLogic conditions are correctly displayed or hidden based on the specified requirements.
- Test B:  Ensure that the survey flow follows the logic defined in `getNextQuestionIdByLogicJump` and that the refactored function handles edge cases 

## Checklist



### Required


### Appreciated


https://github.com/DavinciSalute/formbricks/assets/151516329/82b0e058-c210-4b38-8f9d-aac29b8cc086
